### ISSUE TITLE
@sanity-default-login - unable to replace providers with single sanity.io email/password provider

### ISSUE THREAD

**Issue by mrhut10** [sanity-io/sanity#1702]
**Describe the bug**

Configuring the default-login to replace the defaults with only the one sanity.io login provider, causes the app to crash with a "Uncaught error"

<img width="329" alt="uncaught" src="https://user-images.githubusercontent.com/20330237/72492340-f10ee800-3870-11ea-9dec-c31778fcc097.PNG">

the config file is below, as soon as you change mode from "replace" to "append" then it stops crashing, but then i end up with the defaults plus an extra email/password login method.

//     file: default-login.json
```
{
  "providers": {
    "mode": "replace",
    "redirectOnSingle": true,
    "entries": [
      {
        "name" : "sanity",
        "title" : "E-mail / password",
        "url" : "https://api.sanity.io/v1/auth/login/sanity"
      }
    ]
  }
}
```

**Expected behavior**
i expect sanity to just redirect and default to just the one email/password login.

**Screenshots**
Error:
<img width="329" alt="uncaught" src="https://user-images.githubusercontent.com/20330237/72492340-f10ee800-3870-11ea-9dec-c31778fcc097.PNG">

Configuration
<img width="415" alt="code" src="https://user-images.githubusercontent.com/20330237/72492465-6bd80300-3871-11ea-9201-a6fadc362e89.png">

**Which versions of Sanity are you using?**
@sanity/cli             0.142.2 (latest: 0.147.8)
@sanity/base            0.147.6 (latest: 0.147.8)
@sanity/cli             0.147.3 (latest: 0.147.8)
@sanity/components      0.147.6 (latest: 0.147.9)
@sanity/core            0.147.6 (latest: 0.147.9)
@sanity/default-layout  0.147.6 (latest: 0.147.9)
@sanity/default-login   0.147.0 (up to date)
@sanity/desk-tool       0.147.7 (latest: 0.147.9)

**What operating system are you using?**
Windows 10
**Which versions of Node.js / npm are you running?**
node - v12.1.0
npm - v6.9.0

**Comment by rexxars** [comment]
@mrhut10 I'm not able to reproduce this on 1.148.7 - could you give this a try and let me know if it's still an issue for you?

**Comment by mrhut10** [comment]
I'll have to try, I'll look and get back to you. Ty @rexxars

**Comment by philschonholzer** [comment]
I got the same problem with 1.149.0.

```
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

Stack:

    at checkForNestedUpdates (http://localhost:3333/static/js/vendor.bundle.js:30787:15)
    at scheduleUpdateOnFiber (http://localhost:3333/static/js/vendor.bundle.js:28790:3)
    at Object.enqueueSetState (http://localhost:3333/static/js/vendor.bundle.js:19112:5)
    at LoginDialog.913.Component.setState (http://localhost:3333/static/js/vendor.bundle.js:3663:16)
    at LoginDialog.redirectToProvider (http://localhost:3333/static/js/app.bundle.js:124450:12)
    at LoginDialog.redirectToProvider (http://localhost:3333/static/js/app.bundle.js:267413:25)
    at LoginDialog.componentDidUpdate (http://localhost:3333/static/js/app.bundle.js:124440:12)
    at LoginDialog.componentDidUpdate (http://localhost:3333/static/js/app.bundle.js:267403:123)
    at commitLifeCycles (http://localhost:3333/static/js/vendor.bundle.js:27233:22)
    at commitLayoutEffects (http://localhost:3333/static/js/vendor.bundle.js:30460:7)
```

**Comment by philschonholzer** [comment]
@rexxars if I set `"redirectOnSingle": false` and click on the login button it's written that 

```
This project is missing the required "thirdPartyLogin" feature to support custom logins.
```  



I only want to allow "Email / Password login" provided by Sanity. This should be possible, not?

**Comment by mrhut10** [comment]
yeah pretty much 90% of sites i make clients will only want a email login so complicates it for them if they have multiple choice.

please keep me in the loop if anyone has a solution to this.

**Comment by rexxars** [comment]
A fix was released in v1.149.11 - https://github.com/sanity-io/sanity/releases/tag/v1.149.11
